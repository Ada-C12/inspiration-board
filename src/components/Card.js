import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';

const Card = (props) => {

  const { id, text, deleteCardCallback } = props;
  
  return (
    <div className="card">
      <div className="card__content">
        <p className="card__content-text">{ text }</p>
        { props.emoji !== null ? 
        <p className="card__content-emoji">{ emoji.getUnicode(props.emoji) }</p>
        :
        '' }
      </div>
      <button
        type="button"
        className="card__delete"
        aria-label="Delete"
        onClick={() => { deleteCardCallback(id) }}
      >Take</button>
    </div>
  )
};

Card.propTypes = {
  id: PropTypes.number.isRequired,
  text: PropTypes.string,
  emoji: PropTypes.string,
  deleteCardCallback: PropTypes.func.isRequired,
};

export default Card;
