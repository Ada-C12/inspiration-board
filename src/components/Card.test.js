import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Card from './Card';
describe('Card', () => {
  test('Snapshot test for Card', () => {
    const container = render(
      <Card
        id={5}
        text={'Hooray!'}
        emoji={'strawberry'}
        about={'All about my cat Rufus!'}
        deleteCardCallback={() => { }}
      />);
    expect(container.asFragment()).toMatchSnapshot();
    cleanup();
  });
});
