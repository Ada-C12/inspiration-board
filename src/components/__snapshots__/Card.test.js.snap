// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Snapshot test for Card 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="card__content"
    >
      <div
        class="card__content-text"
      >
        Hooray!
      </div>
      <div
        class="card__content-emoji"
      />
      <button
        class="card__delete"
      >
        Delete
      </button>
    </div>
  </div>
</DocumentFragment>
`;
